lib/datadog/appsec/event.rb

```diff
@@ -52,7 +52,7 @@ def record(span, *events)
           # ensure rate limiter is called only when there are events to record
           return if events.empty? || span.nil?
 
-          Datadog::AppSec::RateLimiter.limit(:traces) do
+          Datadog::AppSec::RateLimiter.instance.limit do
```

Comment:
This line matches Singleton usage and I thought `instance` referred to the only global instance. I suggest another name for it (maybe `local_instance` if you don't have better ideas).

Reply:
Agreed, still working on the name. I don't want to have bypass method `limit` and rather provide builder/access method to the configured instance.

Reply:
I choose the `thread_local` name to reflect locality of the limiter to the client.

```ruby
Datadog::AppSec::RateLimiter.thread_local.limit do
  record_via_span(span, *events)
end

```